pyqtgraph/graphicsItems/DateAxisItem.py

```diff
@@ -213,46 +232,82 @@ def tickValues(self, minVal, maxVal, minSpc):
 ], "99:99:99")
 
 
-def getOffsetFromUtc():
+def fromSecsSinceEpoch(timestamp):
+    try:
+        return QDateTime.fromSecsSinceEpoch(round(timestamp))
+    except OverflowError:
+        return QDateTime()
+
+
+def calculateUtcOffset(timestamp):
+    return -fromSecsSinceEpoch(timestamp).offsetFromUtc()
+
+
+def getPreferredOffsetFromUtc(timestamp, preferred_offset=None):
     """Retrieve the utc offset respecting the daylight saving time"""
-    ts = time.localtime()
-    if ts.tm_isdst:
-        utc_offset = time.altzone
-    else:
-        utc_offset = time.timezone
-    return utc_offset
+    if preferred_offset is not None:
+        return preferred_offset
+    return calculateUtcOffset(timestamp)
+
+
+def adjustTimestampToPreferredUtcOffset(timestamp, offest=None):
+    return timestamp - getPreferredOffsetFromUtc(timestamp, offest)
+
+
+def offsetToLocalHour(timestamp):
+    local = fromSecsSinceEpoch(timestamp)
+    roundedToHour = local.time()
+    roundedToHour.setHMS(roundedToHour.hour(), 0, 0)
+    return -roundedToHour.secsTo(local.time())
+
+
+def shiftLocalTimeToUtcTime(timestamp):
```

Comment:
i still have no idea how to properly name this function

Reply:
`inverseUTCOffset`? That's my first thought

Reply:
Going from UTC to the user timezones is sometimes called "localize". That would make this "delocalize", maybe?

Reply:
I'm good w/ delocalize, but with all naming conventions, I try and conform to what Qt does upstream.

We could expand on Qt's API which has [`offsetFromUtc`](https://doc.qt.io/qt-6/qtimezone.html#offsetFromUtc)

And do:

```python
def applyOffsetFromUtc:
    ...

def applyOffsetToUtc:
    ...
```

Reply:
That seems like a good precendent!